migrate.py

```diff
@@ -213,32 +213,39 @@ def match_import_src(imp_src, import_map):
     raise LookupError(f"Couldn't find a replacement for {imp_src!s}")
 
 
-def rewrite_imports_in_fst(mod_fst, import_map):
+def rewrite_imports_in_fst(mod_fst, import_map, collection, spec):
     """Replace imports in the python module FST."""
+    deps = []
     for imp in mod_fst.find_all(('import', 'from_import')):
         imp_src = imp.value
         if imp.type == 'import':
             imp_src = imp_src[0].value
 
-        # TODO: this assumes the module_utils resides in the same
-        # collection, must use spec to find actual collection and then
-        # both rewrite import AND add collection to depenencies of
-        # current collection.
-        # ansible_collections.jctanner.cloud_vmware.module_utils.
-        # TODO: update gdata.requirements if module_util is in diff
-        # collection
         try:
             token_length, exchange = match_import_src(imp_src, import_map)
-            # TODO: actually lookup part after token in spec to find
-            # 'correct collection'
         except LookupError:
             continue
 
         if len(imp.targets.find_all('name_as_name', value='g:*Base')) > 0:
             continue  # Skip imports of Base classes
 
+        plugin_name = imp_src[-1].value
+        plugin_type = imp_src[-2].value
+        try:
+            plugin_collection = get_plugin_collection(plugin_name, plugin_type, spec)
+        except LookupError:
+            # plugin not in spec, assuming it stays in core and leaving as is
+            continue
+
+        if plugin_collection != collection:
+            deps.append(plugin_collection)
+            # FIXME I am changing a tuple!
+            exchange_new = list(exchange)
+            exchange_new[1] = plugin_collection
+            exchange = tuple(exchange_new)
+
         imp_src[:token_length] = exchange  # replace the import
-    return mod_fst
+    return mod_fst, deps
```

Comment:
Don't really need to return FST, it's modified in-place anyway:
```suggestion
    return deps
```

Reply:
:+1: